tmt/steps/provision/__init__.py

```diff
@@ -2190,6 +2189,25 @@ def mkdtemp(
         else:
             self.execute(Command('rm', '-rf', path))
 
+    @functools.cached_property
+    def sudo_prefix(self) -> str:
```

Comment:
Sounds like an attribute for `GuestFacts`. It's where `is_superuser` or `has_rsync` attributes live, `sudo_prefix` sounds like something fitting that description. And it takes care of logging, catching exceptions in probes, etc.